src/parsing.jl

```diff
@@ -1,156 +1,241 @@
+    isnotcom = occursin("inf", s)
+    if startswith(s, '[') && endswith(s, ']') # parse as interval
+        s = strip(s[2:end-1])
+        if ',' in s # infsupinterval
+            lo, hi = strip.(split(s, ','))
+            isempty(lo) && (lo = "-inf"; isnotcom = true)
+            isempty(hi) && (hi = "inf"; isnotcom = true)
+            lo = parse_num(T, lo, RoundDown)
+            hi = parse_num(T, hi, RoundUp)
+        else # point interval
+
+            (s == "empty" || isempty(s)) && return emptyinterval(T), true # emptyinterval
+            s == "entire" && return entireinterval(T), true # entireinterval
+            lo = parse_num(T, s, RoundDown)
+            hi = parse_num(T, s, RoundUp)
         end
-
-        rhi = dir == 'd' ? Rational(0) : radius
-        rlo = dir == 'u' ? Rational(0) : radius
-        scale = exponent >= 0 ? 10^exponent : 1//10^-exponent
-        x = parse(Int, x)
-        lo = T((x - rlo)*scale, RoundDown)
-        hi = T((x + rhi)*scale, RoundUp)
-        return Interval(lo, hi)
+    elseif '?' in s # uncertainty interval
+        if occursin("??", s) # unbounded interval
+            isnotcom = true
+            m, _ = split(s, "??")
+            if 'u' in s # interval in the form [m, Inf]
+                lo = parse(T, m, RoundDown)
+                hi = T(Inf)
+            elseif 'd' in s # interval in the form [-Inf, m]
+                lo = T(-Inf)
+                hi = parse(T, m, RoundUp)
+            else
+                return entireinterval(T), true
+            end
+        else
+            m , vde = split(s, '?')
+
+            # ulp computation
+            if '.' in m # ulp is last decimal position
+                ulp = 1/(big(10.0)^length(split(m, '.')[2]))
+            else # no decimal, hence ulp is unit
+                ulp = big(1.0)
+            end
+            m = parse(BigFloat, m)
+
+            if 'e' in vde
+                vd, e = split(vde, 'e')
+                e = big(10.0) ^ parse(Int, e)
+            else
+                vd = vde
+                e = big(1.0)
+            end
+
+            if 'u' in vd || 'd' in vd
+                d = last(vd)
+                v = vd[1:end-1]
+            else
+                d = 'b' # both directions
+                v = vd
+            end
+
+            v = isempty(v) ? 1//2 : parse(BigInt, v)
+            if d == 'd'
+                lo = T((m - v * ulp) * e, RoundDown)
+                hi = T(m * e, RoundUp)
+            elseif d == 'u'
+                lo = T(m * e, RoundDown)
+                hi = T((m + v * ulp) * e, RoundUp)
+            else
+                lo = T((m - v * ulp) * e, RoundDown)
+                hi = T((m + v * ulp) * e, RoundUp)
+            end
+        end
+    else # number
+        lo = parse_num(T, s, RoundDown)
+        hi = parse_num(T, s, RoundUp)
     end
+    is_valid_interval(lo, hi) && return Interval(lo, hi), isnotcom
+    throw(ArgumentError("input $s can not be parsed as an interval."))
```

Comment:
Is this equivalent to 
```julia
return interval(lo, hi), isnotcom
```
?

Or is there a case I'm missing?

Reply:
the reason for this is that `_parse` is also called by `parse(DecoratedInterval{Float64}, s)`, using simply `interval` would return `[empty]_trv` instead of `[NaI]` in that case.

Reply:
Understood... Then leave it as it is.